src/com/rakovets/course/challenge/HealingEnemy.java

```diff
@@ -0,0 +1,23 @@
+package com.rakovets.course.challenge;
+
+public class HealingEnemy extends Enemy {
+
+    int power = -1;
+
+    HealingEnemy(int health, String name) {
+        super(health, name);
+    }
+
+    public void attackHero(Hero hero) {
+        System.out.println("*HealingEnemy атакует героя*");
+        hero.health -= 10;
+        power++;
+        this.health += power;
+        if (power > 19) {
+            power = -1;
+        }
+        if (hero.health <= 0) {
+            System.out.println(hero.getName() + " died");
+        }
```

Comment:
Very bad! Because Enemy control health hero -> enemy can set any health for the hero